src/ca/mcgill/cs/jetuml/gui/DiagramCanvasController.java

```diff
@@ -320,6 +326,34 @@ else if(!aSelectionModel.contains(element.get()))
 		}
 	}
 
+	/* Recursively reorder the node to be on top of its parent's all children. 
+	 * If the node is not a child node or it does not have a parent, check if 
+	 * the node is a root node of the diagram and reorder it to be on the top 
+	 * of other root nodes.
+	 */
+	private void reorderSelectedNode(Node pNode) 

```

Comment:
Can't we place this method in `Diagram`? I would also rename to `placeOnTop(Node pNode`)

Reply:
Yes, I was considering to put the method in _Diagram_  before, but I felt that the code only relates to selection handling so I kept it in the _DiagramCanvasController_. Thanks for pointing out! Placing the method in Diagram is indeed better. Resolved it.